article/_posts/2020-06-05-Agent57.md

```diff
@@ -0,0 +1,67 @@
+---
+layout: review
+title: "Agent57: Outperforming the human Atari benchmark"
+tags: reinforcement deep-learning RNN
+author: "Antoine Théberge"
+cite:
+    authors: "Adrià Puigdomènech Badia, Bilal Piot, Steven Kapturowski, Pablo Sprechmann, Alex Vitvitskyi, Daniel Guo, Charles Blundell"
+    title:   "Agent57: Outperforming the Atari Human Benchmark"
+    venue:   "Arxiv"
+pdf: ""
+---
+
+
+# Highlights
+
+- Agent57 is able to perform well on all 57 Atari games in the ALE[^1]
+- Q-function is split in two to decompose the contribution of intrinsic and extrinsic rewards
+- A meta-controller modeled as a bandit problem controls the tradeoff between exploration and exploitation
+
+# Introduction
+
+![](/article/images/Agent57/improvements.svg)
+A condensed history of improvements that led to Agent57
+> source: Deepmind blog
+
+NGU[^2] was a monumental success as it was the first agent to achieve good performance on hard-exploration games. However, it conversely failed to gain good scores on "easier" games. The authors hypothesize that this is because, at each episode, NGU uniformly assign a family of more explorative or exploitative agents at random to gather episodes. This process, according to the authors, is suboptimal as the need for exploration or exploitation varies when learning a game as well as with different games.
+
+# Methods
+
+## State-action value function parameterization
+
+The authors propose to split the state-action value function (Q) into this:
+
+$$Q(x, a, j; \theta) = Q(x,a,j;\theta^e) + \beta_j Q(x,a,h;\theta^i)$$
+
+where $$j$$ is a one-hot vector pointing to the $$(\beta, \gamma)$$ pair controlling exploration and discount for the actor, just like in NGU. The authors show in the appendix that this decomposition is valid and that the optimization of this Q-function is equivalent to the optimization of a single Q-function with reward $$r^e + \beta_j r^i$$.
+
+## Adaptive Exploration over a Family of Policies
+
+NGU uses a family of explorative and exploitative policies during training and trains them all equally, regardless of their actual contributions to the learning process. The authors propose using a meta-controller to select which policies will be used at training and testing time. To do so, the authors propose posing the policy selection as a non-stationary multi-arm bandit problem. At the beginning of each episode, the meta controller selects a bandit "arm" to assign a pair $$(\beta, \gamma)$$ per actor. 
```

Comment:
`(...) which policies will be used at training and testing time.`

Maybe I'm missing something, but does this imply that at test time a *different* policy is used wrt to the one used at train time? I'd assume that the same policy will be used in both, but which policy is used is defined/learned by the meta-controller during training. Am I wrong?

Reply:
The meta-controller also does select which policy will be used at test time. This allows the agent to produce different policies at test time and therefore different behaviors can be had. I'll reword this part to make this clearer.